Merged current version of protege into ros-protege

sasjonge opened the pull request.

matthewhorridge committed:
Update to OWL API 4.5.5

matthewhorridge committed:
Use manual dependency inject from within Protege to get around class loading problems with new OWL API DI mechanism

matthewhorridge committed:
Update to OWL API 4.5.6

matthewhorridge committed:
Export Trove packages for HermiT

matthewhorridge committed:
Upgrade Felix

matthewhorridge committed:
Set version number for 5.5.0-beta-3 release

matthewhorridge committed:
Set version for next development iteration

matthewhorridge committed:
Fixes #765

matthewhorridge committed:
Closes #809

matthewhorridge committed:
Merge branch 'master' of https://github.com/liveontologies/protege

matthewhorridge committed:
Added 'show log' menu item back in

matthewhorridge committed:
Set version number to 5.5.0-beta-4 for release

matthewhorridge committed:
Set version number to 5.5.0-beta-5-SNAPSHOT for next development iteration

matthewhorridge committed:
Changed catalog warning to info

matthewhorridge committed:
Closes #813

matthewhorridge committed:
Fixes #817

matthewhorridge committed:
Bumped version number

matthewhorridge committed:
Bumped version number for beta-6 release

matthewhorridge committed:
Updated version for next development cycle

matthewhorridge committed:
Set version number for beta-7 release

matthewhorridge committed:
Bumped version number to 5.5.0-beta-8-SNAPSHOT

matthewhorridge committed:
Addresses #824

matthewhorridge committed:
Updated version numbers for next release

matthewhorridge committed:
Set the version for next development iteration

matthewhorridge committed:
Closes #558

matthewhorridge committed:
Fixes #656

matthewhorridge committed:
Closes #827

matthewhorridge committed:
Potential fix for #804

matthewhorridge committed:
Fixes #816

matthewhorridge committed:
Closes #249.  This completely removes the broken functionality

matthewhorridge committed:
Added mouse position caching as a partial workaround for #723

This workaround enabled faster browsing through entities, but there is a noticeable delay in showing hyperlinks when mousing over expressions.

matthewhorridge committed:
Better mouse position caching impl

matthewhorridge committed:
Caching workaround for #723

More caching added to improve performance

matthewhorridge committed:
Upgrades the JRE to 8u121

This addresses #702

ykazakov committed:
Fix caching for #723

in MList the method getMousePosition() was cached, but OWLCellRenderer
calls getMousePosition(boolean) that was not cached

matthewhorridge committed:
Merge pull request #832 from liveontologies/master

Fix caching for #723

matthewhorridge committed:
Updating to deal with #714

matthewhorridge committed:
Merge branch 'master' of https://github.com/protegeproject/protege

matthewhorridge committed:
Tidying of Plastic Theme and setting fonts to cover Unicode

matthewhorridge committed:
Removed debugging print out

matthewhorridge committed:
Use better icon for checkbox menu items

matthewhorridge committed:
Better radio button menu items

matthewhorridge committed:
Small tweak to font color

matthewhorridge committed:
Better borders for combo boxes

csnyulas committed:
Fixes #792

matthewhorridge committed:
Closes #835

matthewhorridge committed:
Corrected authorship

matthewhorridge committed:
Added JavaDoc to getCatalogFileMethod

matthewhorridge committed:
Code reformat

matthewhorridge committed:
Changed attribution

matthewhorridge committed:
Reformatted code

matthewhorridge committed:
Added nullable annotation

matthewhorridge committed:
Added nonnull annotations

matthewhorridge committed:
Closes #837

matthewhorridge committed:
Closes #836

matthewhorridge committed:
Use common cell renderer

matthewhorridge committed:
Closes #498

matthewhorridge committed:
Removed unused imports

matthewhorridge committed:
Added cut off limit to FolderGroupManager for addressing #619

matthewhorridge committed:
Better rendering of folder entries

matthewhorridge committed:
Introduce guard condition

matthewhorridge committed:
Remove relativization

matthewhorridge committed:
Be consistent with subfolder vs subdirectory

matthewhorridge committed:
Issue warning for early termination rather than info

matthewhorridge committed:
Stop generating and saving an XML catalog on save (better solution for #619)

matthewhorridge committed:
Reformatted code of OWLModelManagerImpl

matthewhorridge committed:
Added missing nonnull annotation

matthewhorridge committed:
Added Nullable annotations

matthewhorridge committed:
Cleaned up the OntologyCatalogManager a little bit

- Deprecated getActiveCatalog and put in place an alternative method that returns an optional value.

matthewhorridge committed:
Cleaned up OntologyCatalogManager

- Deprecate getRedirect method that returns null and provide an alternative method that returns an optional
- Update code to remove the usage of the deprecated method

matthewhorridge committed:
Tidying of missing imports dialog

matthewhorridge committed:
Closes #839

matthewhorridge committed:
Tweaked style of headers

matthewhorridge committed:
Closes #841

csnyulas committed:
Closes #804

matthewhorridge committed:
Removed System out junk

matthewhorridge committed:
Merge branch 'master' of https://github.com/protegeproject/protege

csnyulas committed:
Closes #829

csnyulas committed:
Closes 842

matthewhorridge committed:
Added extra test for debugging #838

matthewhorridge committed:
Exposed method for testing purposes

csnyulas committed:
Closes 843

csnyulas committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

matthewhorridge committed:
Test case for #838

matthewhorridge committed:
Merge branch 'master' of https://github.com/protegeproject/protege

matthewhorridge committed:
Upgraded the OWL API to fix #747 and #501

matthewhorridge committed:
Addresses #847

matthewhorridge committed:
Fixes #620

matthewhorridge committed:
Better rendering of deprecation strikeout

matthewhorridge committed:
Display disambiguated rendering in view titles

matthewhorridge committed:
Disabmiguate renderings for class names in the class description pane

matthewhorridge committed:
Set the version number for the 5.5.0-beta-9 release

matthewhorridge committed:
Set version numbers for next development iteration

matthewhorridge committed:
Set version number for 5.5.0 release

matthewhorridge committed:
Set version number for next development iteration

matthewhorridge committed:
Upgraded OWL API to 4.5.10

matthewhorridge committed:
Closes #856

matthewhorridge committed:
Added dictionary for language codes

matthewhorridge committed:
Added autovalue libraries

matthewhorridge committed:
Added data structure for lang codes

matthewhorridge committed:
Implements a list of language codes for #784

matthewhorridge committed:
Closes #797

matthewhorridge committed:
Closes #770

matthewhorridge committed:
Closes #691

matthewhorridge committed:
Use Java Optional instead of Guava Optional

matthewhorridge committed:
Switch nullable annotation to nonnullable

matthewhorridge committed:
Removed unused method

matthewhorridge committed:
Addresses #865

matthewhorridge committed:
Closes #10

matthewhorridge committed:
Removed commented out code

matthewhorridge committed:
Disabled expression caching.  This fixes #864 and #637

It is also a workaround for aggressive expression expansion related issues.

matthewhorridge committed:
Keep colon in prefix name

matthewhorridge committed:
Code tidy

matthewhorridge committed:
Set empty prefix (part of a fix for #727)

matthewhorridge committed:
Update empty prefix if ontology IRI changes

matthewhorridge committed:
Fixes #877

matthewhorridge committed:
Added nonnull annotations

matthewhorridge committed:
Use method reference instead of lambda

matthewhorridge committed:
Tidy: Mark fields as final

matthewhorridge committed:
Tidy: Replace entity update for loop with stream

matthewhorridge committed:
Tidy: Added nullable annotation

matthewhorridge committed:
Closes #869

matthewhorridge committed:
Closes #687

Added for all hierarchies

matthewhorridge committed:
Render disambiguated rendering

matthewhorridge committed:
Initial merge entities code

matthewhorridge committed:
Initial UI for merging entities

matthewhorridge committed:
Better UI

matthewhorridge committed:
Added merge strategy

matthewhorridge committed:
Use related synonym for OBO entities (as the safest choice according to the GO docs)

matthewhorridge committed:
Transfer id from merged entity to target entity as has_alternative_id

matthewhorridge committed:
Better label

matthewhorridge committed:
Implements #707

matthewhorridge committed:
Closes #583

matthewhorridge committed:
Fixes #862

matthewhorridge committed:
Implements #861

matthewhorridge committed:
Tidied up Open from URL dialog

matthewhorridge committed:
Implements #881

matthewhorridge committed:
Don't show the stack trace in the log

matthewhorridge committed:
Modernised nav bar

matthewhorridge committed:
Modernised search button

matthewhorridge committed:
Renamed SearchIcon to InlineFieldSearchIcon

matthewhorridge committed:
Nice search icon

matthewhorridge committed:
Nicer nav icons

matthewhorridge committed:
Slightly nicer spacing

matthewhorridge committed:
Add rollover to search icon

matthewhorridge committed:
Modernised toolbar in view banner

matthewhorridge committed:
Modernised entity icons and toolbar icons

matthewhorridge committed:
New look and feel for annotations

matthewhorridge committed:
Don't render quotes around annotation properties in annotations

matthewhorridge committed:
Display ontology from non-active ontologies

matthewhorridge committed:
Place active ontology annotations above others

matthewhorridge committed:
Darker lang tag

matthewhorridge committed:
Don't show string datatypes as it's clutter

matthewhorridge committed:
Upgrade OWL API to 4.5.11

This addresss #849.  It also addresses slow selection behavior (exhibited with GO-edit).

matthewhorridge committed:
Support for OBO Foundry Registry

matthewhorridge committed:
Adds link extraction for OBO Foundry Identifiers.  Implements #880

matthewhorridge committed:
Resolve using identifiers.org before purl IRI

matthewhorridge committed:
Use bold font for imported ontology IRI

matthewhorridge committed:
Implements #883

matthewhorridge committed:
Render metrics using thousands separator

matthewhorridge committed:
Use ComponentFactory

Commit:
Add identitas-j as a dependency to protege-owl pom.xml

matthewhorridge committed:
Don't disambiguate rendering in expressions

Commit:
add identitas button to the Newentitiespreferencespanel file

Nizal-Shammry committed:
Create a new file to run randomProlong from identitas Util

Nizal-Shammry committed:
Rename IdentitasRandomProlong.java to RandomProlong.java

matthewhorridge committed:
Initial support for user Id ranges parsed from the idranges.owl ODK files

matthewhorridge committed:
More tests for IdPolicy parser

matthewhorridge committed:
Renamed

matthewhorridge committed:
Renamed

matthewhorridge committed:
Throw an exception if ranges are not correctly formatted

matthewhorridge committed:
Renamed method

matthewhorridge committed:
Updated to throw exxceptions rather than siliently ignore

matthewhorridge committed:
Ranmed exception

matthewhorridge committed:
Renamed enum

matthewhorridge committed:
Handling for automatically setting id ranges

matthewhorridge committed:
Added UI for choosing range when current user does not match any user name in id ranges policy

matthewhorridge committed:
Fixes #664

matthewhorridge committed:
Fixes #855

csnyulas committed:
Fixes #886.

matthewhorridge committed:
Ui clean up

Nizal-Shammry committed:
requested changes are made on RandomProlong

Changes are as follow:
1. Remove RandomProlong constructor because it does the same code as in the getNextID method which is irrelevant.
2. Remove the e.printStackTrace() and use a logger (slf4j) to log exceptions.
3. Remove the checkpoint() and revert() methods because they do nothing. 
4. Explain why using the getRawID method, because we implement the AbstractIDGenerator interface and we had to implement all of its methods.
5. Finally, remove the replace("-", "_") statement that was in the code and keep the id generated with dashes in between. The reason for replacing the dashes with underscores in the first place, because I noticed in the UniqueIdGenerator class they used underscores and I thought this preferable in protege.

Nizal-Shammry committed:
correct typo message in throw exception

matthewhorridge committed:
Merge branch 'master' of https://github.com/protegeproject/protege

matthewhorridge committed:
Closes #368

matthewhorridge committed:
Upgraded OWL API to 4.5.12

Fixes #887
Fixes #888

matthewhorridge committed:
Changed layout of literal editor

matthewhorridge committed:
Code tidy.  Don't bother to enable/disable fields.

matthewhorridge committed:
Rename datatype field

matthewhorridge committed:
Code reformat

matthewhorridge committed:
Modernized IRI editor

matthewhorridge committed:
Use entity finder field for entity IRI

matthewhorridge committed:
Moved error message to below lexical field

matthewhorridge committed:
Nicer language tag editing field

matthewhorridge committed:
Renamed lang tag field

matthewhorridge committed:
Changed order of datatype and lang tag fields

matthewhorridge committed:
Better matching for autocomplete on lang codes

matthewhorridge committed:
Auto complete on double click

matthewhorridge committed:
Created SuggestField

matthewhorridge committed:
Factored out suggest field

matthewhorridge committed:
Changed parser to produce decimal (rather than integer) and string (rather than plain literal)

matthewhorridge committed:
Clean up

matthewhorridge committed:
Added autoscroll to suggest field

matthewhorridge committed:
Don't set datatype if it's a plain literal

matthewhorridge committed:
Added see other redirect code

matthewhorridge committed:
Fixes #897

matthewhorridge committed:
Proper names for ontology prefixes table

matthewhorridge committed:
Code tidy

matthewhorridge committed:
Code tidy: Factor out methods from long case block

matthewhorridge committed:
Code tidy: Use guard clause instead of surrounding whole code block

matthewhorridge committed:
Removed sorting as this causes the table to jump

matthewhorridge committed:
Tidied logging

matthewhorridge committed:
Code Tidy: Use guard clause

matthewhorridge committed:
Sort on refill

matthewhorridge committed:
Don't allow toolbar to be floated

Locke committed:
fix typo in warnUserIfReasonerIsNotConfigured

csnyulas committed:
Merge pull request #899 from Locke/patch-1

fix typo in warnUserIfReasonerIsNotConfigured

csnyulas committed:
Fixed typos in capitalization of the Reasoner menu

Fixing additional typo that was pointed out in the #899 pull request.

matthewhorridge committed:
Use MList for prefixes

matthewhorridge committed:
Support vertical group layout

matthewhorridge committed:
Changed prefixes view to use MList

matthewhorridge committed:
Impl of clear()

matthewhorridge committed:
Commit changes to prefixes

matthewhorridge committed:
Clear deleted mappings

matthewhorridge committed:
Sort by prefix name

matthewhorridge committed:
Support range validation for annotation properties

matthewhorridge committed:
Merge branch 'master' of https://github.com/protegeproject/protege

matthewhorridge committed:
Updates to work with latest identifiers.org

matthewhorridge committed:
Tidied code

johardi committed:
Fixes #905

matthewhorridge committed:
Added period after OBSOLETE to address #909

matthewhorridge committed:
Init fields with Optional.empty

matthewhorridge committed:
Fixes #910

matthewhorridge committed:
Switch to openjdk8

matthewhorridge committed:
Move xvfb to services

matthewhorridge committed:
Better readme text

matthewhorridge committed:
Added code of conduct

matthewhorridge committed:
Merge pull request #913 from protegeproject/add-code-of-conduct-1

Added code of conduct

matthewhorridge committed:
Adding contributing.md

matthewhorridge committed:
Initial draft

matthewhorridge committed:
Updated issue tracker link

matthewhorridge committed:
Fixed second issues link

matthewhorridge committed:
Added link to contributing guide

matthewhorridge committed:
Possible fix for casting problem mentioned in #910

matthewhorridge committed:
Moved classes in core module to core packages to address #915

matthewhorridge committed:
Made AddChildIcon implement HasUseSystemForeground for uniformity

matthewhorridge committed:
Merge pull request #884 from Nizal-Shammry/Identitas

Provides Identitas IRIs in Protege

matthewhorridge committed:
More efficient memory usage for entity rendering cache

matthewhorridge committed:
Fixes #919

Commit:
Merged current version of protege and ros-protege.

daniel86 commented:
cool someone finally did it :)

Have you checked on whether the RDF/XML export is still producing identical output?

sasjonge commented:
I tested the new version as follows:

- I checked that loading the knowrob.owl (which I haven't loaded before) will resolve the ros-style imports. This works.
- I opened the knowrob.owl and added a class. Some changes in the file did occur:
   - The order of the imports and namespaces in the OWL file change (seemingly in alphabetical order). No imports and namespaces are lost or added.
   -  It changed an item that had the form
` <owl:ObjectProperty rdf:about="http://knowrob.org/kb/knowrob.owl#hasLifetime"> </owl:ObjectProperty>`
to 
` <owl:ObjectProperty rdf:about="http://knowrob.org/kb/knowrob.owl#hasLifetime"/>`
   - And it simply added the class I added in protege to the file.

I also played around with the addition/removal of subclasses, properties and subproperties, but beside the already noted changes, I couldn't find anything problematic.
